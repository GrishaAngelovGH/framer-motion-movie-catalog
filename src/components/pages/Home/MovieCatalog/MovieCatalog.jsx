import { motion } from "framer-motion"

import movies from "./movies.json"

const MovieCatalog = () => {
  const cols = []
  const allMovies = [...movies]

  while (allMovies.length) {
    const col = allMovies.splice(0, 4)
    cols.push(col)
  }

  return (
    <div className="bg-blue-800 min-h-full flex flex-col md:flex-row justify-center items-center">
      {
        cols.map((items, i) => (
          <motion.div key={i} className="flex flex-col" initial={{ x: `-${i + 1}00%` }} animate={{ x: "0%" }} transition={{ duration: 3, delay: 1 }}>
            {
              items.map(v => (
                <img key={v.id} src={`images/${v.image}`} className="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5" />
              ))
            }
          </motion.div>
        ))
      }
    </div>
  )
}

export default MovieCatalog