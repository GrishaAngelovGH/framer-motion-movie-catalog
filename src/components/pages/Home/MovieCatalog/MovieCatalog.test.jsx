import { render } from "@testing-library/react"

import MovieCatalog from "./MovieCatalog"

test("should render MovieCatalog component", () => {
  const view = render(<MovieCatalog />)

  expect(view).toMatchSnapshot()
})