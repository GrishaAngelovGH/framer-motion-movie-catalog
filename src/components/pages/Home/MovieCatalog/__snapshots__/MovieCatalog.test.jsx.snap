// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render MovieCatalog component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-blue-800 min-h-full flex flex-col md:flex-row justify-center items-center"
      >
        <div
          class="flex flex-col"
          style="transform: translateX(-100%) translateZ(0);"
        >
          <img
            class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
            src="images/the-garfield-movie.jpg"
          />
          <img
            class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
            src="images/fly-me-to-the-moon.jpg"
          />
          <img
            class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
            src="images/kung-fu-panda-4.jpg"
          />
          <img
            class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
            src="images/kingdom-of-the-planet-of-the-apes.jpg"
          />
        </div>
        <div
          class="flex flex-col"
          style="transform: translateX(-200%) translateZ(0);"
        >
          <img
            class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
            src="images/transformers-one.jpg"
          />
          <img
            class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
            src="images/mufasa-the-lion-king.jpg"
          />
          <img
            class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
            src="images/beverly-hills-cop-axel-f.jpg"
          />
          <img
            class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
            src="images/indiana-jones-and-the-dial-of-destiny.jpg"
          />
        </div>
        <div
          class="flex flex-col"
          style="transform: translateX(-300%) translateZ(0);"
        >
          <img
            class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
            src="images/gran-turismo.jpg"
          />
          <img
            class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
            src="images/aquaman-and-the-lost-kingdom.jpg"
          />
          <img
            class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
            src="images/the-super-mario-bros-movie.jpg"
          />
          <img
            class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
            src="images/spider-man-across-the-spider-verse.jpg"
          />
        </div>
        <div
          class="flex flex-col"
          style="transform: translateX(-400%) translateZ(0);"
        >
          <img
            class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
            src="images/shooting-stars.png"
          />
          <img
            class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
            src="images/sweetwater.jpg"
          />
          <img
            class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
            src="images/a-million-miles-away.png"
          />
          <img
            class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
            src="images/the-marvels.jpg"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-blue-800 min-h-full flex flex-col md:flex-row justify-center items-center"
    >
      <div
        class="flex flex-col"
        style="transform: translateX(-100%) translateZ(0);"
      >
        <img
          class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
          src="images/the-garfield-movie.jpg"
        />
        <img
          class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
          src="images/fly-me-to-the-moon.jpg"
        />
        <img
          class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
          src="images/kung-fu-panda-4.jpg"
        />
        <img
          class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
          src="images/kingdom-of-the-planet-of-the-apes.jpg"
        />
      </div>
      <div
        class="flex flex-col"
        style="transform: translateX(-200%) translateZ(0);"
      >
        <img
          class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
          src="images/transformers-one.jpg"
        />
        <img
          class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
          src="images/mufasa-the-lion-king.jpg"
        />
        <img
          class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
          src="images/beverly-hills-cop-axel-f.jpg"
        />
        <img
          class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
          src="images/indiana-jones-and-the-dial-of-destiny.jpg"
        />
      </div>
      <div
        class="flex flex-col"
        style="transform: translateX(-300%) translateZ(0);"
      >
        <img
          class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
          src="images/gran-turismo.jpg"
        />
        <img
          class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
          src="images/aquaman-and-the-lost-kingdom.jpg"
        />
        <img
          class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
          src="images/the-super-mario-bros-movie.jpg"
        />
        <img
          class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
          src="images/spider-man-across-the-spider-verse.jpg"
        />
      </div>
      <div
        class="flex flex-col"
        style="transform: translateX(-400%) translateZ(0);"
      >
        <img
          class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
          src="images/shooting-stars.png"
        />
        <img
          class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
          src="images/sweetwater.jpg"
        />
        <img
          class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
          src="images/a-million-miles-away.png"
        />
        <img
          class="rounded-lg md:w-[160px] lg:w-[220px] min-h-[350px] m-5"
          src="images/the-marvels.jpg"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
